Foundation/Process.swift

```diff
@@ -978,4 +979,127 @@ private func posix(_ code: Int32) {
     default: fatalError("POSIX command failed with error: \(code)")
     }
 }
+
+#if os(Android)
+// Android doesn't provide posix_spawn APIs until recent API level, so we cannot
+// depend on them, but we can imitate the API, and perform the same work.
+
+private struct posix_spawn_file_actions_t {
+    enum Action {
+    case dup2(old: Int32, new: Int32)
+    case close(fd: Int32)
+    }
+
+    private(set) var actions: [Action] = []
+
+    mutating func add(_ action: Action) {
+        actions.append(action)
+    }
+}
+
+private struct posix_spawnattr_t {}
+
+private func posix_spawn_file_actions_init(_ actions: inout posix_spawn_file_actions_t) -> Int32 {
+    // NOTE: Nothing to do here, the struct initializes itself.
+    return 0
+}
+
+@discardableResult
+private func posix_spawn_file_actions_destroy(_ actions: inout posix_spawn_file_actions_t) -> Int32 {
+    // NOTE: Nothing to do here, the struct handles its own memory.
+    return 0
+}
+
+private func posix_spawn_file_actions_adddup2(_ actions: inout posix_spawn_file_actions_t, _ old: Int32, _ new: Int32) -> Int32 {
+    actions.add(.dup2(old: old, new: new))
+    return 0
+}
+
+private func posix_spawn_file_actions_addclose(_ actions: inout posix_spawn_file_actions_t, _ fd: Int32) -> Int32 {
+    actions.add(.close(fd: fd))
+    return 0
+}
+
+private func posix_spawn(_ pidPtr: UnsafeMutablePointer<pid_t>?,
+                         _ launchPath: UnsafePointer<Int8>?,
+                         _ actions: UnsafeMutablePointer<posix_spawn_file_actions_t>?,
+                         _ attrp: UnsafeMutablePointer<posix_spawnattr_t>?,
+                         _ argv: UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>?,
+                         _ envp: UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>?) -> Int32 {
+    // TODO: We completely ignore attrp, because at the moment, the only
+    // invocation doesn't pass a value.
+
+    // Block signals during this fork/execv dance.
+    var signalSet = sigset_t()
+    sigfillset(&signalSet)
+    var oldmask = sigset_t()
+    guard sigprocmask(SIG_BLOCK, &signalSet, &oldmask) == 0 else {
+        fatalError("sigprocmask() call failed: \(errno)")
```

Comment:
[sigprocmask](https://pubs.opengroup.org/onlinepubs/7908799/xsh/sigprocmask.html) should not fail in this case, since the only failure case is an incorrect first parameter, which is hardcoded here to a valid value. If the code ends up in the `fatalError`, something has gone really wrong, and aborting is the right thing to do, in my opinion.

Reply:
Ah right, I see `sigprocmask` only returns `EINVAL` and `EFAULT` as errors which does imply a programming error.